### PR TITLE
Add a confirmation for recovery of account

### LINKED ISSUE #645: Add new confirmation dialog for password reset
There have been enough instances in the past few months of people accidentally resetting their password (without realizing that deletes all data) that we should add an additional confirmation prompt, making it clear the data will be gone and it'll be irreversible.

### PR DESCRIPTION
**Pull request by BrunoBernardino**
Basically make it clearer the data will be deleted.

Fixes #645

Also fix typo.

### CODE REVIEW

File: packages/app/src/elements/recover.ts [hunk 1]
```diff
@@ -185,7 +185,18 @@ export class Recover extends StartForm {
             }
         }
 
-        return this._recover(email, password);
+        const doRecover = await confirm(
+            $l("Your data **WILL BE DELETED** and cannot be recovered!"),
```
Comment (MaKleSoft) [inline packages/app/src/elements/recover.ts:189]: Minor change in wording considering that shared vaults are not affected.

```suggestion
            $l("All data in your personal vault **WILL BE DELETED** and cannot be recovered!"),
```

File: packages/app/src/elements/recover.ts [hunk 2]
```diff
@@ -185,7 +185,18 @@ export class Recover extends StartForm {
...
+            $l("Delete data and recover account"),
+            $l("Cancel"),
+            {
+                icon: "warning",
```
Comment (MaKleSoft) [inline packages/app/src/elements/recover.ts:193]:
```suggestion
                icon: "warning",
                type: "destructive",
```

**Review by MaKleSoft — CHANGES_REQUESTED** [review]
Had some minor changes but looks good otherwise.

File: packages/app/src/elements/recover.ts (hunk 1 above)
Comment (BrunoBernardino) [reply packages/app/src/elements/recover.ts:189]: While technically more correct, I think this becomes too wordy and prone to be ignored as the first warnings. Do you really prefer your version?

Comment (MaKleSoft) [reply packages/app/src/elements/recover.ts:189]: It's a bit more wordy, but I think it's fine, since they still have to click a big red button (`type: "destructive"` will turn the button red) with the words "Delete data" on it.
